cluster-autoscaler/cloudprovider/mcm/mcm_manager.go

```diff
@@ -411,6 +435,11 @@ func (m *McmManager) SetMachineDeploymentSize(machinedeployment *MachineDeployme
 			return err
 		}
 
+		// don't scale down during rolling update, as that could remove ready node with workload
+		if md.Spec.Replicas >= int32(size) && !isRollingUpdateFinished(md) {
+			return fmt.Errorf("MachineDeployment %s is under rolling update , cannot reduce replica count", md.Name)
+		}
+
```

Comment:
Same with this. Concerning scale down, this method is called by `DecreaseTargetSize`, which is called by `fixNodeGroupSize`.

Reply:
same reason